rewrite-core/src/main/java/org/openrewrite/Paint.java

```diff
@@ -15,13 +15,28 @@
  */
 package org.openrewrite;
 
-import com.fasterxml.jackson.annotation.JsonIdentityInfo;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import com.fasterxml.jackson.annotation.JsonTypeInfo;
-import com.fasterxml.jackson.annotation.ObjectIdGenerators;
+import org.openrewrite.internal.lang.Nullable;
 
-@Incubating(since = "2.0.0")
-@JsonIdentityInfo(generator = ObjectIdGenerators.IntSequenceGenerator.class, property = "@ref")
-@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, property = "@c")
-public interface Metadata {
+/**
+ * Mark any AST element with "paint". Used by search visitors to mark AST elements that
+ * match the search criteria. By painting AST elements in a tree, search results can be
+ * contextualized in the tree that they are found in.
+ */
+@Incubating(since = "7.0.0")
+public class Paint implements Markers.Marker {
```

Comment:
I picture a world where we'll want to compose search visitors, hereafter called "searchers". 
Imagine a searcher that paints every usage of a particular symbol and a searcher that detects a particular usage pattern, like anything that has the potential to throw an NPE. If they compose then you could easily describe "Find every usage of this symbol that could lead to an NPE". 

But since a collection of Markers can only have one marker of any given type, I worry that the existence of a public `Paint` type is an invitation to write searchers visitors that wont be able to compose together. The unit tests running any given searcher in isolation would pass, but as soon as two searchers are applying `Paint` you're stuck with "OR" semantics when maybe you wanted "AND". 

If we're sticking with the "Markers can only have one Marker per type of Marker" constraint, and I agree that constraint is likely to keep interacting with that API simple, then we may want to diligently avoid creating _any_ `Marker` implementation that isn't owned by a particular searcher. _Especially_ one that's public in the core libraries that multiple different people in different organizations might independently chose to use.

Reply:
Making `Paint` an interface or giving it a `protected` constructor might also work, instead of deleting it.

Reply:
I think you've convinced me to remove the "only one type per markers" constraint. Some marker types should be distinct, others not. Hmm...